### PR TITLE
feat(terraform): add support for terraform_version in tfe_workspace blocks

### LINKED ISSUE #12251: Enable upgrading of terraform_version for terraform cloud workspaces
### What would you like Renovate to be able to do?

Terraform Cloud has a [terraform module](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/workspace#terraform_version) with a `terraform_version` setting that allows to set the terraform version used.

This is currently not detected by renovate and solved by us using a regexManager configuration:

```json
{
  "regexManagers": [
    {
      "fileMatch": [
        "\\.tf$"
      ],
      "depNameTemplate": "hashicorp/terraform",
      "datasourceTemplate": "github-tags",
      "extractVersionTemplate": "v(?<version>.*)$",
      "matchStrings": [
        "renovate\\s+terraform_version\\s+=\\s\"(?<currentValue>.*)\""
      ]
    }
  ]
}
```

Where the terraform file looks like this:

```hcl
resource "tfe_workspace" "test_workspace" {
  name              = "test"
  organization      = "testorg"
  execution_mode    = "local"
  working_directory = "terraform/test_workspace"

  # renovate
  terraform_version = "1.0.2"
}
```

### If you have any ideas on how this should be implemented, please tell us here.

For all terraform files, check for the `terraform_version` option inside of all `resource "tfe_workspace"` blocks.

I’m not sure if this is a feature to specific to one provider, therefore I’m opening this issue to get feedback in before trying to implement it.

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by morremeyer**
## Changes

Implements support for the `terraform_version` parameter in `tfe_workspace` resources.

## Context

This resolves #12251 and enables one more dependency to be updated without the need for a regexManager. 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by secustor — CHANGES_REQUESTED** [review]
Thx, for your contribution!

I have some small remarks, but nothing big.

File: lib/modules/manager/terraform/extract.spec.ts [hunk 1]
```diff
@@ -17,6 +17,8 @@ const helm = loadFixture('helm.tf');
 const lockedVersion = loadFixture('lockedVersion.tf');
 const lockedVersionLockfile = loadFixture('rangeStrategy.hcl');
 const terraformBlock = loadFixture('terraformBlock.tf');
+const tfeWorkspaceBlock = loadFixture('tfeWorkspaceTerraformVersion.tf');
+const tfeWorkspaceNoVersion = loadFixture('tfeWorkspaceNoTerraformVersion.tf');
```
Comment (secustor) [inline lib/modules/manager/terraform/extract.spec.ts:21]: Combine this two file, to test the actual code.   
Currently it is aborting during the pre-flight check. https://github.com/renovatebot/renovate/pull/14522/files#diff-5ab62e69287f1f4f2c26b5d60412284ac7ca35b03655fdd19a126da08d7ba944R49

**Review by secustor — CHANGES_REQUESTED** [review]

File: lib/modules/manager/terraform/extract.spec.ts (hunk 1 above)
Comment (morremeyer) [reply lib/modules/manager/terraform/extract.spec.ts]: That makes so much sense, thanks for pointing it out. That actually also caught a bug.

**Comment by morremeyer** [comment]
I set `no-version` as the skip reason if there is no `terraform_version` , it could probably also be `empty`. Do you have an opinion about that?

**Review by secustor — APPROVED** [review]
`no-version` seems to fit here well. 

LGTM

**Review by rarkins — APPROVED** [review]

**Pull request merged**